services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crossregion/S3CrossRegionSyncClient.java

```diff
@@ -47,4 +52,43 @@ protected <T extends S3Request, ReturnT> ReturnT invokeOperation(T request, Func
     private void handleOperationFailure(Throwable t, String bucket) {
         //TODO: handle failure case
     }
+
+    @SuppressWarnings("unchecked")
+    private <T extends S3Request> T requestWithDecoratedEndpointProvider(T request, String bucket) {
+        return (T) request.toBuilder()
+                          .overrideConfiguration(getOrCreateConfigWithEndpointProvider(request, bucket))
+                          .build();
+    }
+
+    //TODO: optimize shared sync/async code
```

Comment:
Do we have a backlog task in case we forget?

Reply:
I didn't think we would since it's quite annoying and I'm pretty sure we wouldn't forget, but just in case I added a task.